securesystemslib/signer/_aws_signer.py

```diff
@@ -101,136 +116,132 @@ def import_(cls, aws_key_id: str, local_scheme: str) -> Tuple[str, Key]:
         Raises:
             UnsupportedAlgorithmError: If the AWS KMS signing algorithm is
             unsupported.
-            BotoCoreError: Errors from the botocore.exceptions library.
+            BotoCoreError: Errors from the botocore library.
             ClientError: Errors related to AWS KMS client.
         """
         if AWS_IMPORT_ERROR:
             raise UnsupportedLibraryError(AWS_IMPORT_ERROR)
 
-        client = boto3.client("kms")
-        request = client.get_public_key(KeyId=aws_key_id)
+        if local_scheme and local_scheme not in cls.aws_signing_algorithms:
+            raise ValueError(f"Unsupported scheme: {local_scheme}")
+
+        try:
+            client = boto3.client("kms")
+            request = client.get_public_key(KeyId=aws_key_id)
+        except (BotoCoreError, ClientError) as e:
+            logger.error(
+                "Failed to import key using AWS KMS key ID %s: %s",
+                aws_key_id,
+                str(e),
+            )
+            raise e
+
+        keytype = cls._get_keytype_from_aws_response(request)
+        aws_scheme = request["SigningAlgorithms"][0]
+
+        if not local_scheme:
+            if keytype == "ecdsa":
+                local_scheme = cls._get_ecdsa_scheme(aws_scheme)
+            elif keytype == "rsa":
+                local_scheme = "rsassa-pss-sha256"
+            else:
+                raise ValueError(f"Unsupported key type: {keytype}")
+
+        if keytype == "ecdsa" and local_scheme != cls._get_ecdsa_scheme(
+            aws_scheme
+        ):
+            raise ValueError(
+                f"The AWS KMS key does not support the scheme: {local_scheme}"
+            )
+
+        if (
+            keytype == "rsa"
+            and local_scheme not in cls.aws_signing_algorithms.values()
+        ):
+            raise ValueError(
+                f"The AWS KMS key does not support the scheme: {local_scheme}"
+            )
```

Comment:
Let's remove these checks after all. And add them after #612...

```suggestion
```

Reply:
ah okay, i can i add what i mentioned above after https://github.com/secure-systems-lab/securesystemslib/issues/612.